### PR TITLE
Promisify Villager & Trader

### PR DESCRIPTION
**Pull request by ph0t0shop**
Aiming to promisify the last villager stuff, including api, ts bindings, and test

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1703,9 +1703,12 @@ you are opening.
 #### bot.openVillager(villagerEntity)
 
 Returns an `Villager` instance which represents the trading window you are opening.
+You can listen to the `ready` event on this `Villager` to know when it's ready
 
 #### bot.trade(villagerInstance, tradeIndex, [times], [cb])
 
+This function also returns a `Promise`, with `void` as its argument upon completion.
```
Comment (wvffle) [inline docs/api.md:1710]: There are many javascript developers that are unfamiliar with void concept.

**Review by wvffle — CHANGES_REQUESTED** [review]

File: docs/api.md (hunk 1 above)
Comment (wvffle) [reply docs/api.md:1710]: And it's not an argument, rather a resolved value.

Comment (ph0t0shop) [reply docs/api.md:1710]: I just mean that it's a void promise, there is no argument type

Comment (ph0t0shop) [reply docs/api.md:1710]: All the other promise functions also contain this line

**Review by rom1504 — APPROVED** [review]

**Pull request merged**